pkg/internal/kubebuilder/validation/dns.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// This file's code was modified from "k8s.io/apimachinery/pkg/util/validation"
+// to avoid package dependencies. In case of additional functionality from
+// "k8s.io/apimachinery" is needed, re-consider whether to add the dependency.
+
+const (
```

Comment:
Hi @joelanford the validations here we could use directly from `k8s.io/apimachinery/pkg/util/validation` instead of that. WDYT?  Kubebuilder has a politic to avoid add deps and it was the reason for the validations be replicated to the project. However, I think would be better to use the validation from API machinery in order to avoid it gets outdated.

Reply:
Fixed